src/modeling/core/coefficients.jl

```diff
@@ -197,6 +229,24 @@ function evaluate_coefficient(coeff::LVCoordinateSystemCache, geometry_cache::Ce
     return LVCoordinate(x1, x2, x3)
 end
 
+# GPU coefficient evaluation!
+function evaluate_coefficient(coeff::LVCoordinateSystemCache, geometry_cache::FerriteUtils.DeviceCellCache, qv::FerriteUtils.StaticQuadratureValues{T}, t) where {T}
```

Comment:
You can try to add some union type for the device and normal cell caches and see if JET gives you green light on it.

```suggestion
function evaluate_coefficient(coeff::LVCoordinateSystemCache, geometry_cache::FerriteUtils.DeviceCellCache, qv::FerriteUtils.StaticQuadratureValues{T}, t) where {T}
```

Reply:
still both have different arguments (QuadraturePoint, StaticQuadratureValues) and also different function for `shape_value` 
```julia
function evaluate_coefficient(coeff::LVCoordinateSystemCache, geometry_cache::CellCache, qp::QuadraturePoint{ref_shape,T}, t) where {ref_shape,T}
    @unpack cv, cs = coeff
    @unpack dh     = cs
    x1 = zero(T)
    x2 = zero(T)
    x3 = zero(T)
    dofs = celldofsview(dh, cellid(geometry_cache))
    @inbounds for i in 1:getnbasefunctions(cv)
        val = shape_value(cv, qp, i)::T ##### here
        x1 += val * cs.u_transmural[dofs[i]]
        x2 += val * cs.u_apicobasal[dofs[i]]
        x3 += val * cs.u_rotational[dofs[i]]
    end
    return LVCoordinate(x1, x2, x3)
end

# GPU coefficient evaluation!
function evaluate_coefficient(coeff::LVCoordinateSystemCache, geometry_cache::FerriteUtils.DeviceCellCache, qv::FerriteUtils.StaticQuadratureValues{T}, t) where {T}
    @unpack cv, cs = coeff
    @unpack dh     = cs
    x1 = zero(T)
    x2 = zero(T)
    x3 = zero(T)
    dofs = celldofsview(dh, FerriteUtils.cellid(geometry_cache))
    @inbounds for i in 1:getnbasefunctions(cv)
        val = FerriteUtils.shape_value(qv, i)::T ####### here
        x1 += val * cs.u_transmural[dofs[i]]
        x2 += val * cs.u_apicobasal[dofs[i]]
        x3 += val * cs.u_rotational[dofs[i]]
    end
    return LVCoordinate(x1, x2, x3)
end

```